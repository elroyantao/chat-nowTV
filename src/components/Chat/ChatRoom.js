import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { compareByTimestamp } from '../../helpers/time-helpers'
import ChatMessage from './ChatMessage'

class ChatRoom extends Component {
  static propTypes = {
    messages: PropTypes.arrayOf(PropTypes.object)
  }
  render() {
    const { messages } = this.props
    if (!messages) return null
    return (
      <div>
        {messages.map(message => (
          <ChatMessage key={message.id} message={message}/>
        ))}
      </div>
    )
  }
}

const messageSortSelector = (state) => {
  return state.messages.sort(compareByTimestamp)
}

const mapStateToProps = state => {
  return {
    messages: messageSortSelector(state)
  };
};

export default connect(
  mapStateToProps,
  null
)(ChatRoom);