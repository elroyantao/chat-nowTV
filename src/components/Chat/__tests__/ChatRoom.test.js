import testHelper from '../../../helpers/test-helper'
import ChatRoom from '../ChatRoom'

describe('<ChatRoom/>', () => {
  const renderComponent = testHelper(ChatRoom.WrappedComponent)
  describe('@renders', () => {
    it('in default state', () => {
      expect(renderComponent().getTree()).toMatchSnapshot()
    })
    it('when messages are defined', () => {
      const props = {
        messages: [{ id: 123 }, { id: 32425 }, { id: 923 }]
      }
      const { wrapper, getTree } = renderComponent(props)
      expect(getTree()).toMatchSnapshot()
      expect(wrapper.find('Connect(ChatMessage)').length).toBe(3)
    })
  })
})