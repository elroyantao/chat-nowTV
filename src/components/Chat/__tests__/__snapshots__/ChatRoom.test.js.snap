exports[`<ChatRoom/> @renders in default state 1`] = `""`;

exports[`<ChatRoom/> @renders when messages are defined 1`] = `
<div>
  <Connect(ChatMessage)
    key="123"
    message={
      Object {
        "id": 123,
      }
    } />
  <Connect(ChatMessage)
    key="32425"
    message={
      Object {
        "id": 32425,
      }
    } />
  <Connect(ChatMessage)
    key="923"
    message={
      Object {
        "id": 923,
      }
    } />
</div>
`;
